### ISSUE TITLE
Mismatch with `opt_einsum` on FLOPs counting

### ISSUE THREAD

**Issue by mofeing** [bsc-quantic/EinExprs.jl#29]
I'm not sure why there is this mismatch but I've found the number of FLOPs reported by `EinExprs` is approximately half of what `opt_einsum` reports.

I believe that it might be because we are only counting floating-point multiplications, and not sums.

If that is the case, should we really consider counting sums? Current implementations use `fmad` instructions (i.e. Fused Multiply And Add) which fuse multiplication and addition in only one instruction.

**Comment by samuelsonric** [comment]
> the costs opt_einsum reports assumes a real datatype for the tensors, which is generally twice as high compared to generic ‘ops’ (i.e. [`tree.contraction_cost()`](https://cotengra.readthedocs.io/en/latest/autoapi/cotengra/index.html#cotengra.ContractionTree.contraction_cost)) and four times lower than assuming a complex datatype (i.e. [`tree.total_flops('complex')`](https://cotengra.readthedocs.io/en/latest/autoapi/cotengra/index.html#cotengra.ContractionTree.total_flops)).

https://cotengra.readthedocs.io/en/latest/contraction.html#differences-with-opt-einsum

**Comment by mofeing** [comment]
True, but like `opt_einsum`, we assume all tensors to be of the same eltype, so the mismatch should be by a multiplicative factor which is not what I saw. 🥲
